nix/binary.nix

```diff
@@ -0,0 +1,99 @@
+{ callPackage
+, cmake
+, darwin
+, helix-core-api-set
+, hostPlatform
+, http-parser
+, lib
+, libiconv
+, openssl
+, overrideSDK
+, patchelf
+, pcre
+, pkg-config
+, stdenv
+, zlib
+}:
+let
+  inherit (callPackage ./util.nix { }) unNixifyDylibs;
+  # at the time of writing, there is an SDK version issue when building on
+  # x86_64-darwin, so we force it to 11.0.
+  # in future nixpkgs version, SDK versions should be better and more
+  #  consistently configured.
+  stdenv' = (if stdenv.isDarwin then overrideSDK stdenv "11.0" else stdenv);
+in
+unNixifyDylibs (stdenv'.mkDerivation rec {
+  name = "p4-fusion";
+  version = "v1.13.2-sg";
```

Comment:
any chance we could somehow sync this string with the constant in code? 😇

Reply:
Was thinking that, maybe a text file that gets included in the binary. Turns out thats not really a thing in C++ land outside of shenanigans like `objcopy` :clueless: